### PR TITLE
Load video assets as needed

### PR DESCRIPTION
**Pull request by rxl881**

### CODE REVIEW

File: src/js/components/TourismScene.js [hunk 1]
```diff
@@ -60,8 +80,9 @@ export default class TourismScene extends React.Component {
         this.transformUvs(0);
 
         // Play the video and set up event listeners
-        this.videos[this.state.videoIndex].play();
-        this.videos[this.state.videoIndex].addEventListener('ended', this.playNextVideo);
+        this.videos[newIndex].currentTime = 0;
+        this.videos[newIndex].play();
+        this.videos[newIndex].addEventListener('ended', this.playNextVideo);
```
Comment (superdump) [inline src/js/components/TourismScene.js]: Add the event listener before playing imo.

File: src/js/components/TourismScene.js [hunk 2]
```diff
@@ -110,12 +131,15 @@ export default class TourismScene extends React.Component {
     }
 
     playNextVideo() {
+        // console.warn("This videos: ", this.videos, this.state.videoIndex);
```
Comment (superdump) [inline src/js/components/TourismScene.js]: Stray debug log.

File: src/js/components/structures/Playlist.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+limitations under the License.
+*/
+
+
+import React from 'react';
+
+export default class Playlist extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { loadIndex: -1};
```
Comment (superdump) [inline src/js/components/structures/Playlist.js]: Stray space after `{`

File: src/js/components/structures/Playlist.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    render() {
+        return null;
+    }
+}
+
+Playlist.propTypes = {
+    playlistId: React.PropTypes.string.isRequired,
+    items: React.PropTypes.arrayOf(React.PropTypes.shape({
```
Comment (superdump) [inline src/js/components/structures/Playlist.js]: Dodgy indentation on the following lines.

File: src/js/components/structures/Playlist.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    componentWillUnmount() {
+        document.querySelectorAll('#assets #' + this.props.playlistId + ' video').forEach(function(video){
+            video.removeEventListener('play', this.loadNext());
+        }.bind(this));
+    }
+
+    _createElements() {
+        this.props.items.forEach(function(item, index) {
```
Comment (superdump) [inline src/js/components/structures/Playlist.js]: Arrow function?

File: src/js/components/structures/Playlist.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        this.assets = document.querySelector('#assets #' + this.props.playlistId);
+        this.loadNext = this.loadNext.bind(this);
+    }
+
+    componentDidMount() {
+        this._createElements();
+    }
+
+    componentWillUnmount() {
+        document.querySelectorAll('#assets #' + this.props.playlistId + ' video').forEach(function(video){
```
Comment (superdump) [inline src/js/components/structures/Playlist.js]: Arrow function?

File: src/js/components/structures/Playlist.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+            this.setState({
+                loadIndex: loadIndex,
+            });
+            video.addEventListener('play', this.loadNext);
+        } else {
+            console.log('All playlist items loaded');
+        }
+    }
+
+    render() {
```
Comment (superdump) [inline src/js/components/structures/Playlist.js]: This shouldn't be a react component if it's not going to return anything. It should just be a ES6 class.

**Comment by rxl881** [comment]
Closing as this is replaced by a (hopefully) improved approach in #28.

**Pull request closed without merging**